models/models.js

```diff
@@ -0,0 +1,29 @@
+const db = require('../db/connection.js')
+
+function fetchTopics() {
+    return db.query("SELECT * FROM topics")
+        .then((data) => {
+            return data.rows
+        })
+}
+
+function fetchArticles() {
+    return db.query("SELECT * FROM articles")
+        .then((data) => {
+            return data.rows
+        })
+}
+
+function fetchArticle(num) {
+    return db.query("SELECT * FROM articles")
+        .then((data) => {
+            if (!data.rows[num]) return Promise.reject();
```

Comment:
This needs changing; if i request article number 3, rows should still have a length of one and therefore rows[3] wouldn't exist but it's still a valid request. Its best to check if there is anything at rows[0] or check the length of rows instead